python/sparkdl/param/image_params.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Some parts are copied from pyspark.ml.param.shared and some are complementary
+to pyspark.ml.param. The copy is due to some useful pyspark fns/classes being
+private APIs.
+"""
+
+from pyspark.ml.param import Param, Params, TypeConverters
+from pyspark.sql.functions import udf
+
+from sparkdl.image.imageIO import imageArrayToStruct, imageSchema
+from sparkdl.param import SparkDLTypeConverters
+
+OUTPUT_MODES = ["vector", "image"]
+
+class HasTensorOutput(Params):
+
+    outputMode = Param(Params._dummy(), "outputMode",
+                       "How the output column should be formatted. 'vector' for a 1-d MLlib " +
+                       "Vector of floats. 'image' to format the output to work with the image " +
+                       "tools in this package.",
+                       typeConverter=SparkDLTypeConverters.supportedNameConverter(OUTPUT_MODES))    
+
+class HasKerasModel(Params):
+
+    modelFile = Param(Params._dummy(), "modelFile",
+                      "HDF5 file containing the Keras model (architecture and weights)",
+                      typeConverter=TypeConverters.toString)
+
+    kerasParams = Param(Params._dummy(), "kerasParams",
+                        "dict with parameters passed to Keras model fit method")
+
+
+    def setModelFile(self, value):
+        return self._set(modelFile=value)
+
+    def getModelFile(self):
+        return self.getOrDefault(self.modelFile)
+
+    def setKerasParams(self, value):
+        return self._set(kerasParams=value)
+    
+    def getKerasParams(self):
+        return self.getOrDefault(self.kerasParams)
+
+
+class CanLoadImage(Params):
+
+    imageLoader = Param(Params._dummy(), "imageLoader",
+                        "Function containing the logic for loading and pre-processing images. " +
+                        "The function should take in a URI string and return a 4-d numpy.array " +
+                        "with shape (batch_size (1), height, width, num_channels).")
+
+    def setImageLoader(self, value):
+        return self._set(imageLoader=value)
+
+    def getImageLoader(self):
+        return self.getOrDefault(self.imageLoader)
+
+    def _loadedImageCol(self):
+        return "__sdl_img"
+    
+    def _loadImages(self, dataset, inputCol):
```

Comment:
this is not really private but protected. I would rather name it without a `_` and mention in the doc that is internal, or call it `loadImagesInternal`